### DISCUSSION TITLE
Remove default CSP header content partly

### DISCUSSION THREAD

**Discussion by dargmuesli** [Baroshem/nuxt-security#280] [Q&A] [answered]
Hey, is there any way to remove the default `script-src-attr` from the CSP headers? I know that I can remove the `'none'` value from it by settings its value to `[]` but that still keeps the key around, potentially making values in `script-src` ineffective.

**Comment by Baroshem** [comment]
Hey,

Have you tried:

```ts
  security: {
    headers: {
      contentSecurityPolicy: {
        'script-src-attr': false,
      }
    },
  }
```

**Reply by dargmuesli** [reply]
Oh, I think I had another webserver running, `script-src-attr: []` seems to work contrary to my previous testing. `false` does not work as `CSPSourceValue` does not include it.

**Reply by Baroshem** [reply]
Ahh I see. So is it ok right now?

Regarding the second part. I think this will show you TS error but will still work as I have tested it in my test app.

We can add it to the type declaration

**Reply by dargmuesli** [reply]
~~Yes, just verified it, `script-src-attr: []` removes the header part for me :)~~

**Reply by dargmuesli** [reply]
Nvm, `script-src-attr: []` does remove the `script-src-attr` part **on development**, but not on **static build**!
That's where my confusion came from.

**Reply by dargmuesli** [reply]
`'script-src-attr': false` actually does result in the header being removed in development as well as static build.

**Reply by dargmuesli** [reply, marked as answer]
I raised https://github.com/Baroshem/nuxt-security/pull/286 to address this

**Comment by Baroshem** [comment]
Closing as the feature was implemented :)
